modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/account_selector/OrdersTable.js

```diff
@@ -13,12 +13,12 @@
  */
 
 import ClayTable from '@clayui/table';
+import DateRenderer from 'frontend-taglib-clay/data_set_display/data_renderers/DateRenderer';
+import StatusRenderer from 'frontend-taglib-clay/data_set_display/data_renderers/StatusRenderer';
```

Comment:
It feels hacky to me to import inner components of a taglib module. How often do we do this? :thinking: 
- If this is rare, maybe we can refactor to work around this? For example, we can extract `DateRenderer` logic to a string util in `frontend-js-web`. For `StatusRenderer` we can simply duplicate code, or move to `liferay-js-components-web`.
- If this is common to do, we should make a more comprehensive solution. This needs some analysis, but my initial idea would be to have all `*Renderer`s in a common module, either a new module like `frontend-js-data-renderers-web`, or `liferay-js-components-web`. This doesn't look that easy as we are [referencing local utils](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/data_renderers/ActionLinkRenderer.js#L22), so we need to move them as well. There is also a question of what to do with [DataSetDisplayContext](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/data_renderers/ActionLinkRenderer.js#L23) in some of them. This goes beyond the scope of this PR.

Reply:
Hey @markocikos!

The quirkiness of such import is probably due to the fact that `frontend-taglib-clay` contains "too much", both in terms of full JS implementation, utils, events constants and taglib exposure and, as you say, these renderers (as well as many other parts of the `DataSetDisplay`) should probably be properly split across multiple new/existing modules or simply cleaned up of duplicated code already present in LFR.

For the moment being I simply wanted to reconcile the usages of those renderers within our `commerce-frontend-js` module not to break anything, but I reckon those 2 renderers are definitely simple and can be handled as per your _first_ proposal, avoiding out-of-the-guts imports like this. 😄 

Even more, the `StatusRenderer` could be simply reduced to the base `ClayLabel` with utility classes, as that's what that component is actually rendering.

Also, consider those renderers are currently used in just 1 component of ours (`AccountSelector`) and nowhere else.

What do you prefer? Should I change these things in this PR or send you an immediate follow-up?

Reply:
@gianmarcobrunialti I like _out-of-the-guts_ expression, captures the case well :D But yeah, we are exposing internal structure of imported module instead of using its top-level API, and that's not good. Even worse in this case is we importing from a `taglib` module, that should contain only internal logic of tags, with API in `*.tld` file.

I think we are on the same page for the proper solution, with split of entire `DataSetDisplay` across correct modules. But it's a big, maybe even epic-level effort, outside of scope of this task. For this task, let's just try to work around gut imports.

>Even more, the StatusRenderer could be simply reduced to the base ClayLabel with utility classes, as that's what that component is actually rendering.

Sounds great! :+1: 

`DateRenderer` can be moved to a string utility.

`getDataRendererById` looks problematic, as it exposes all `Renderer`s. Would it be possible to refactor `AddOrCreate` not to use it? Should this item finder maybe be a part of `DataSetDisplay`? :thinking:  

> What we cannot do for the time being is to remove them, as both are heavily used/driven by BE configurations for DataSetDisplay's spread all around.

I am not sure what you mean. `Renderer`s are perfectly fine as internal mechanism of `DataSetDisplay`, and as an abstraction in related configurations. There is no need to remove them. But, the situation changes if we want to use `Renderer` components as general-purpose components, like they are used in this PR. In this case, we want to either put them in a correct place, or refactor solution not to use them.

>What do you prefer? Should I change these things in this PR or send you an immediate follow-up? 

Whatever makes more sense to you. If the changes are small / medium, feel free to change them in this PR. If they scope up to a new task, we can push the temporary ugliness and fix it in a new task or tasks.